### PR TITLE
Initial Burrow EVM TP

### PR DESCRIPTION
**Pull request by aludvik**
This implements a basic, first-pass attempt at creating a Burrow EVM TP. The TP can be built and can talk to state and arguments can be passed to the EVM through a simple payload.

Additional work to be done:
- Improve the efficiency of accessing storage. Possibilities include spreading storage across multiple addresses, implementing a cache at the TP level, or storing some data in a separate TP database.
- Setup an event system
- Design and implement an efficient way to pass EVM return values to a client
- Reorganize the build environment

### CODE REVIEW

**Comment by aludvik** [comment]
Testing:
1. Re-provision vagrant or just run `tools/plugins/install_go_tools.sh` again.
2. Run `build_go`
3. Startup a validator and rest_api
4. Startup the Burrow EVM TP with `tp_burrow_evm -vv`
5. Run the client test script `python3 families/burrow_evm/src/sawtooth_burrow_evm/client/client.py`

File: families/burrow_evm/src/sawtooth_burrow_evm/client/client.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+import yaml
+
+import sawtooth_signing as signing
+
+import sawtooth_sdk.protobuf.evm_pb2 as evm_pb2
+import sawtooth_sdk.protobuf.transaction_pb2 as transaction_pb2
+import sawtooth_sdk.protobuf.batch_pb2 as batch_pb2
+
+def sha(d):
```
Comment (peterschwarz) [inline families/burrow_evm/src/sawtooth_burrow_evm/client/client.py]: this should be named `sha512`

File: protos/evm.proto [hunk 2]
```diff
@@ -0,0 +1,56 @@
+// Copyright 2017 Intel Corporation
```
Comment (vaporos) [inline protos/evm.proto]: I thought the only time we said that was really true was in the examples directory.

File: families/burrow_evm/src/burrow/evm/vm.go [hunk 3]
```diff
@@ -113,15 +102,17 @@ func NewVM(appState AppState, params Params, origin Word256, txid []byte) *VM {
...
-	v, err := acc.Permissions.Base.Get(perm)
-	if _, ok := err.(ptypes.ErrValueNotSet); ok {
-		if appState == nil {
-			// In this case the permission is unknown
-			return false
-		}
-		return HasPermission(nil, appState.GetAccount(ptypes.GlobalPermissionsAddress256), perm)
-	}
-	return v
+	return true
```
Comment (vaporos) [inline families/burrow_evm/src/burrow/evm/vm.go]: Would prefer to track this todo in jira and remove the commented out code.

File: families/burrow_evm/src/sawtooth_burrow_evm/handler/sawtooth_app_state.go [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+import (
+	. "burrow/evm"
+	"burrow/evm/sha3"
+	. "burrow/word256"
+	"github.com/golang/protobuf/proto"
+	"sawtooth_sdk/processor"
+	. "sawtooth_sdk/protobuf/evm_pb2"
+)
+
+// Ideas:
```
Comment (vaporos) [inline families/burrow_evm/src/sawtooth_burrow_evm/handler/sawtooth_app_state.go]: Could we track this in jira?

File: families/burrow_evm/src/sawtooth_burrow_evm/client/client.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        payloadDict = yaml.safe_load(fin.read())
+else:
+    print("Please specify YAML to load")
+    exit(1)
+
+private_key = signing.generate_privkey()
+public_key = signing.generate_pubkey(private_key)
+
+namespace = sha('burrow-evm')[0:6]
+print(namespace)
```
Comment (vaporos) [inline families/burrow_evm/src/sawtooth_burrow_evm/client/client.py]: debugging statement?

**Comment by aludvik** [comment]
FYI, I am reworking a lot of this after finishing the Spec.

**Pull request closed without merging**